ARCHITECTURE.md

```diff
@@ -1,113 +1,280 @@
- - The Host will invoke either `GameNetPortal.StartHost` or `StartUnityRelayHost` if Unity Relay is being used.
- - The client will invoke either `ClientGameNetPortal.StartClient` or `StartClientUnityRelayModeAsync`.
- - Boss Room's own connection validation logic is performed in `ServerGameNetPortal.ApprovalCheck`, which is plugged in to the `NetworkManager`'s connection approval callback. Here, some basic information about the connection is recorded (including a GUID, to facilitate future reconnect logic), and success or failure is returned. In the future, additional game-level failures will be detected and returned (such as a `ServerFull` scenario). 
+For MainMenu scene we only have the `MainMenuClientState`, however for the scenes that contain networked logic we also have the `server` counterparts to the client scenes, and both exist on the same game object.
 
-## Data model
-Game data in Boss Room is defined in `ScriptableObjects`. The `ScriptableObjects` are organized by enum and made available in a singleton class: the `GameDataSource`, in particular `ActionDescription` and `CharacterData`. `Actions` represent discrete verbs (like swinging a weapon, or reviving someone), and are substantially data driven. Characters represent both the different player classes, and also monsters, and represent basic details like health, as well as what "Skill" Actions are available to each Character.
+As soon as we get into the CharSelect scene (either by joining or hosting a game) - our NetworkManager instance is running. The host drives game state transitions and also controls the set of scenes that are currently loaded in the game - this indirectly forces all the clients to load the same set of scenes as the server they are connected to (via Netcode's networked scene management).
+
+### Application Flow Diagram
+
+![boss room scene flow](Documentation/Images/BossRoomSceneFlowDiagram.png "Boss Room Scene Flow")
+
+> __Note__:
+>
+> The main room's scene is split in four scenes where the primary scene (BossRoom's root scene) contains the State components and game logic, along with the navmesh for the level and the trigger areas that let the server know that it needs to load a given subscene. Each subscene is then loaded additively with those triggers.
+>
+> Subscenes contain spawn points for the enemies and visual assets for their respective segment of the level. The server unloads subscenes that don't contain any active players and then loads the subscenes that are needed based on the position of the players - if at least one player overlaps with the subscene's trigger area, the subscene is loaded.
 
 ## Transports
-Currently two network transport mechanisms are supported: 
+
+Currently two network transport mechanisms are supported:
+
 - IP based
+
 - Unity Relay Based
 
 In the first, the clients connect directly to a host via IP address. This will only work if both are in the same local area network or if the host forwards ports.
```

Comment:
```suggestion
When using IP, clients connect directly to a host via an IP address. This will only work if both client and host are in the same local area network or if the host forward ports.
```